graphql_compiler/tests/schema_transformation_tests/test_rename_schema.py

```diff
@@ -186,6 +261,34 @@ def test_original_unmodified_suppress(self):
         rename_schema(original_ast, {"Human": None})
         self.assertEqual(original_ast, parse(ISS.multiple_objects_schema))
 
+    def test_rename_illegal_noop_unused_renaming(self):
+        with self.assertRaises(NoOpRenamingError) as e:
+            rename_schema(parse(ISS.basic_schema), {"Dinosaur": "NewDinosaur"})
+        self.assertTrue(check_no_op_renaming_error_message({"Dinosaur"}, set(), e.exception))
+
+    def test_rename_legal_noop_unused_renaming(self):
+        # Unlike with test_rename_illegal_noop_unused_renaming, here the renaming is not
+        # iterable. As a result, this renaming is inadvisable but it is technically legal to
+        # write a renaming like this since the intended "Dinosaur" -> "NewDinosaur" mapping is
+        # unused and will silently do nothing when applied to the given schema.
+        class RenameMapping(object):
+            def __init__(self):
+                pass
+
+            def get(self, key, default=None):
```

Comment:
Need type hints here, and probably a docstring. I wonder why the linter isn't complaining? Could you look into that?

Reply:
Turns out the mypy.ini has `disallow_untyped_defs = False` for the tests and uncommenting that uncovered roughly 1 complaint per test. For now, I'll just type-hint this and separately open a PR type-hinting the rest